drivers/flash/flash_npcx_fiu_qspi.c

```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright (c) 2023 Nuvoton Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nuvoton_npcx_fiu_qspi
+
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/flash/npcx_flash_api_ex.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/spi.h>
+#include <zephyr/dt-bindings/flash_controller/npcx_fiu_qspi.h>
+#include <soc.h>
+
+#include "flash_npcx_fiu_qspi.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(npcx_fiu_qspi, LOG_LEVEL_ERR);
+
+/* Driver convenience defines */
+#define HAL_INSTANCE(dev) \
+	((struct fiu_reg *)((const struct npcx_qspi_fiu_config *)(dev)->config)->base)
+
+/* Timeout for UMA transaction. (Unit:ms) */
+#define NPCX_UMA_TIMEOUT 10
```

Comment:
Oops. It is a definition we don't use. I already implemented a timeout mechanism based on dt property `max-timeout` in flash_npcx_nor_wait_until_ready() function. In a special case, we might not get a ready bit set from its status register if there is no flash on qspi bus. (For example, no external nor spi flash mounted on the board) Then, it will cause a watchdog timeout and reboot the system. I will remove this line later,

Reply:
Done.